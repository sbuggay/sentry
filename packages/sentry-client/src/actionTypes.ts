export const INITIALIZE = "INITIALIZE";

export const SERVER_ADD = "SERVER_ADD";
export const SERVER_REMOVE = "SERVER_REMOVE";
export const SERVER_EDIT = "SERVER_EDIT";
export const SERVER_POLL = "SERVER_POLL";
export const SERVER_POLL_SUCCESS = "SERVER_POLL_SUCCESS";
export const SERVER_POLL_FAILURE = "SERVER_POLL_FAILURE";

export const VIEW_CHANGE = "VIEW_CHANGE";

export const STATE_LOAD = "STATE_LOAD";
export const STATE_SAVE = "STATE_SAVE";

export const UPDATE_LAST_UPDATED = "UPDATE_LAST_UPDATED";
