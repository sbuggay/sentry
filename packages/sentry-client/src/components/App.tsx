import * as React from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

import List from "./List";
import LastUpdated from "./LastUpdated";
import Legend from "./Legend";
import Storage from "./Storage";
import GroupSelect from "./GroupSelect";

import { EView } from "../constants";

import { initialize, addServer, changeView } from "../actions";

interface IStateProps {
    title?: string;
    view?: EView;
}

interface IDispatchProps {
    initialize?: () => any;
    addServer?: (values: object) => any;
    changeView?: (view: EView) => void;
}

export class App extends React.Component<IStateProps & IDispatchProps, any> {

    public componentWillMount() {
        if (this.props.initialize) {
            this.props.initialize();
        }
    }

    public onServerInputSubmit(values: object) {
        if (this.props.addServer) {
            this.props.addServer(values);
        }
    }

    public getStyle() {
        return {
            maxWidth: "900px",
            marginLeft: "auto",
            marginRight: "auto"
        };
    }

    public handleSelectChange(index: EView) {
        if (this.props.changeView) {
            this.props.changeView(index);
        }
    }

    public render() {

        const options = [];
        for (const option in EView) {
            options.push(EView[option]);
        }

        return (
            <div>
                <h1 style={{ textAlign: "center", fontWeight: 500 }}>
                    {this.props.title}
                </h1>
                <div style={this.getStyle()}>
                    <div style={{ display: "flex", justifyContent: "space-between" }}>
                        <div></div>
                         <GroupSelect
                            onChange={(index) => this.handleSelectChange(index)}
                            options={options}
                            selectedOption={this.props.view || EView.servers} />
                    </div>
                    <List />
                    <div style={{ display: "flex", justifyContent: "space-between" }}>
                        <LastUpdated />
                        <Legend />
                    </div>
                    <div>
                        <Storage />
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state: any) => {
    return {
        title: state.title,
        view: state.view
    };
};

const mapDispatchToProps = (dispatch: any) => {
    return bindActionCreators({
        addServer,
        initialize,
        changeView
    }, dispatch);
};

export default connect<IStateProps & IDispatchProps, any, any>(mapStateToProps, mapDispatchToProps)(App);
