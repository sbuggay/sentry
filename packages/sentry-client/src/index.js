"use strict";
var React = require("react");
var ReactDOM = require("react-dom");
var redux_1 = require("redux");
var react_redux_1 = require("react-redux");
var redux_thunk_1 = require("redux-thunk");
var App_1 = require("./sentry/components/App");
var rootReducer_1 = require("./rootReducer");
var store = redux_1.createStore(rootReducer_1.default, redux_1.applyMiddleware(redux_thunk_1.default));
ReactDOM.render(React.createElement(react_redux_1.Provider, { store: store },
    React.createElement(App_1.default, null)), document.getElementById("app"));
//# sourceMappingURL=index.js.map