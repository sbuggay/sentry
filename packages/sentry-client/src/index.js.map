{"version":3,"file":"index.js","sourceRoot":"","sources":["index.tsx"],"names":[],"mappings":";AAEA,6BAA+B;AAC/B,oCAAsC;AACtC,+BAAqD;AACrD,2CAAuC;AACvC,2CAAgC;AAEhC,+CAA0C;AAE1C,6CAAwC;AAExC,IAAM,KAAK,GAAG,mBAAW,CACxB,qBAAW,EACX,uBAAe,CAAC,qBAAK,CAAC,CACtB,CAAC;AAEF,QAAQ,CAAC,MAAM,CACd,oBAAC,sBAAQ,IAAC,KAAK,EAAE,KAAK;IACrB,oBAAC,aAAG,OAAG,CACG,EACX,QAAQ,CAAC,cAAc,CAAC,KAAK,CAAC,CAC9B,CAAC"}