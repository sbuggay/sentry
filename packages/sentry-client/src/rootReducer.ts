import reducer from "./reducer";

// This file exists as a future entry point for mulitple stores, currently this
// is not needed.

export default reducer;
