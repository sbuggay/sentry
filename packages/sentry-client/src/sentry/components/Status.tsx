import * as React from "react";

import { STATUS } from "../constants/status";

interface IStatusProps {
	status: any;
}

export class Status extends React.Component<IStatusProps, any> {
	render() {
		let className, style;

		switch (this.props.status) {
			case STATUS.AVAILABLE:
				className = "fa fa-circle";
				style = { color: "#229926" };
				break;
			case STATUS.ISSUE:
				className = "fa fa-exclamation-circle";
				style = { color: "#FEE032" };
				break;
			case STATUS.MAINTENANCE:
				className = "fa fa-square";
				style = { color: "#1172C6" };
				break;
			default:
			case STATUS.OUTAGE:
				className = "fa fa-exclamation-triangle";
				style = { color: "#951A1D" };
				break;
		}

		return <i className={className} style={style} aria-hidden="true"></i>;
	}
}

export default Status;
