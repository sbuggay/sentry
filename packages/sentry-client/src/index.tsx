// require("file-loader?name=[name].[ext]!../dist/index.html");

import * as React from "react";
import * as ReactDOM from "react-dom";
import { createStore, applyMiddleware, compose } from "redux";
import { Provider } from "react-redux";
import thunk from "redux-thunk";

import App from "./sentry/components/App";

import rootReducer from "./rootReducer";

const store = createStore(
    rootReducer,
    applyMiddleware(thunk)
);

ReactDOM.render(
    <Provider store={store}>
        <App />
    </Provider>,
    document.getElementById("app")
);