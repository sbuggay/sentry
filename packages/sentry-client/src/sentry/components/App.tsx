import * as React from "react";
import { connect } from "react-redux";

import ServerList from "./ServerList";
import ServerForm from "./ServerForm";
import Status from "./Status";

import { addServer } from "../actions";

import { guid } from "../lib/utils";

import { STATUS } from "../constants/status";

interface IDispatchProps {
	addServer?: Function;
};

class App extends React.Component<IDispatchProps, any> {
	onServerInputSubmit(values: Object) {
		console.log(values);
		this.props.addServer(values);
	}

	render() {
		return <div>
			<Status status={STATUS.AVAILABLE} />
			<Status status={STATUS.ISSUE} />
			<Status status={STATUS.OUTAGE} />
			<Status status={STATUS.MAINTENANCE} />
			<ServerForm handleSubmit={this.onServerInputSubmit.bind(this)} />
			<ServerList />
		</div>;
	}
}

const mapDispatchToProps = (dispatch: any) => {
	return {
		addServer: (values: any) => {
			dispatch(addServer(values.nameValue, values.hostValue, guid()));
		}
	};
};

export default connect<any, IDispatchProps, any>(undefined, mapDispatchToProps)(App);