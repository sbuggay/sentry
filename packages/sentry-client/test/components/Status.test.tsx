import * as React from "react";
import * as renderer from "react-test-renderer";
import { Status } from "../../src/components/Status";
import { EStatus } from "../../src/constants";

describe("Status", () => {
    it("renders an available status", () => {
        const component = renderer.create(
            <Status status={EStatus.available} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders an outage status", () => {
        const component = renderer.create(
            <Status status={EStatus.outage} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders an issue status", () => {
        const component = renderer.create(
            <Status status={EStatus.issue} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders a maintenance status", () => {
        const component = renderer.create(
            <Status status={EStatus.maintenence} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
