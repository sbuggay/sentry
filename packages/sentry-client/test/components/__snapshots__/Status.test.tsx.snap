// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status renders a maintenance status 1`] = `
<i
  aria-hidden="true"
  className="fa fa-square"
  style={
    Object {
      "color": "#1172C6",
      "marginRight": "0.5em",
    }
  }
/>
`;

exports[`Status renders an available status 1`] = `
<i
  aria-hidden="true"
  className="fa fa-circle"
  style={
    Object {
      "color": "#62C552",
      "marginRight": "0.5em",
    }
  }
/>
`;

exports[`Status renders an issue status 1`] = `
<i
  aria-hidden="true"
  className="fa fa-exclamation-circle"
  style={
    Object {
      "color": "#fadf59",
      "marginRight": "0.5em",
    }
  }
/>
`;

exports[`Status renders an outage status 1`] = `
<i
  aria-hidden="true"
  className="fa fa-exclamation-triangle"
  style={
    Object {
      "color": "#DF3E2D",
      "marginRight": "0.5em",
    }
  }
/>
`;
